src/mongo/delegates/SearchDocuments.class.php

```diff
@@ -40,133 +45,130 @@ public function __construct($storeName, Collection $collection, $defaultContext,
-        if(empty($searchSpec)){
+        if (empty($searchSpec)) {
             $this->debugLog("Could not find Search Document Specification for $specId");
             return null;
         }
 
-        if (isset($searchSpec["from"]))
-        {
-            $from = $searchSpec["from"];
-        }
-        else
-        {
+        if (isset($searchSpec['from'])) {
+            $from = $searchSpec['from'];
+        } else {
             $from = $this->podName;
         }
 
         // work out whether or not to index at all
         $proceedWithGeneration = false;
 
-        foreach ($searchSpec['filter'] as $indexRules)
-        {
+        foreach ($searchSpec['filter'] as $indexRules) {
             // run a query to work out
-            if (!empty($indexRules['condition']))
-            {
+            if (!empty($indexRules['condition'])) {
                 $irFrom = (!empty($indexRules['from'])) ? $indexRules['from'] : $this->podName;
                 // add id of current record to rules..
-                $indexRules['condition']['_id'] = array(
+                $indexRules['condition']['_id'] = [
                     'r'=>$this->labeller->uri_to_alias($resource),
-                    'c'=>$this->labeller->uri_to_alias($context));
+                    'c'=>$this->labeller->uri_to_alias($context)
+                ];
 
-                if ($this->getConfigInstance()->getCollectionForCBD($this->storeName, $irFrom)->findOne($indexRules['condition']))
-                {
+                if ($this->getConfigInstance()->getCollectionForCBD($this->storeName, $irFrom)->findOne($indexRules['condition'])) {
                     // match found, add this spec id to those that should be generated
-                   $proceedWithGeneration = true;
+                    $proceedWithGeneration = true;
                 }
-            }
-            else
-            {
+            } else {
                 // no restriction rules, so just add to generate
                 $proceedWithGeneration = true;
             }
         }
 
-        if($proceedWithGeneration == false){
-            $this->debugLog("Unable to proceed with generating $specId search document for $resource, does not satisfy rules");
+        if ($proceedWithGeneration == false) {
+            $this->debugLog(
+                "Unable to proceed with generating $specId search document for $resource, does not satisfy rules"
+            );
             return null;
         }
 
-        $_id = array(
+        $_id = [
             'r'=>$this->labeller->uri_to_alias($resource),
             'c'=>$this->labeller->uri_to_alias($context)
-        );
+        ];
 
-        $sourceDocument = $this->getConfigInstance()->getCollectionForCBD($this->storeName, $from)->findOne(array('_id'=>$_id));
+        // build the document, set the timestamp before we fetch the source document
+        $generatedDocument = [\_CREATED_TS => DateUtil::getMongoDate()];
```

Comment:
This line is the only code change (and that's just moved from line 114) in this file.